pkg/metrics/statsd/cosmosdb/metrics.go

```diff
@@ -0,0 +1,90 @@
+package cosmodb
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/metrics"
+)
+
+var _ http.RoundTripper = (*tracerRoundTripper)(nil)
+
+type tracerRoundTripper struct {
+	m   metrics.Interface
+	tr  http.RoundTripper
+	log *logrus.Entry
+}
+
+func New(log *logrus.Entry, tr *http.Transport, m metrics.Interface) *tracerRoundTripper {
+	return &tracerRoundTripper{
+		m:   m,
+		tr:  tr,
+		log: log,
+	}
+}
+
+func (t *tracerRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	start := time.Now()
+	resp, err := t.tr.RoundTrip(r)
+	if resp == nil {
+		return resp, err
+	}
+
+	action := "unknown"
```

Comment:
var action, table, code string
var ru float64
- or better, use `:=` where they're actually assigned to for real